content/6_Advanced/Count_Min.mdx

```diff
@@ -3,38 +3,227 @@ id: count-min
-**none** of the input rectangles.
+### Implementation
+
+<LanguageSection>
+
+<CPPSection>
+
+```cpp
+const int MAXN = 2e5;
+const int INF = INT32_MAX;
+
+struct Node {
+	int val = INF, cnt = 1;
+} Tree[2 * MAXN];
+
+// combines two segment tree nodes
+Node merge(Node a, Node b) {
+	if (a.val < b.val) {
+		return a;
+	} else if (a.val > b.val) {
+		return b;
+	}
+	return {a.val, a.cnt + b.cnt};
+}
+
+// updates the ith value to v
+void update(int i, int v) {
+	for (Tree[i += MAXN] = {v, 1}; i > 1; i >>= 1) {
+		Tree[i >> 1] = merge(Tree[i], Tree[i ^ 1]);
+	}
+}
+
+// returns the minimum and occurences between indices l and r
+Node query(int l, int r) {
+	Node res;
+	for (l += MAXN, r += MAXN + 1; l < r; l >>= 1, r >>= 1) {
+		if (l & 1) res = merge(res, Tree[l++]);
+		if (r & 1) res = merge(res, Tree[--r]);
+	}
+	return res;
+}
+```
+
+</CPPSection>
+
+</LanguageSection>
+
+### Solution - Area of Rectangles
+
+<Spoiler title = "Hint 1">
+
+Count the number of squares that are covered by **none** of the input rectangles.
+
+</Spoiler>
+
+<Spoiler title = "Hint 2">
+
+Sort the rectangles by x-coordinate and run a sweep-line.
 
 </Spoiler>
 
 <Spoiler title="Solution">
 
-<IncompleteSection>
+We can use techniques introduced in [Range Queries with Sweep Line](https://usaco.guide/plat/range-sweep?lang=cpp)
+
+We sweep from left to right over the $x$-coordinates, maintaining two events for each rectangle: one for the left boundary and one for the right boundary. Maintain a Lazy Segment Tree over the $y$-coordinates. 
+- When we run into a left boundary of some rectangle with $y$-coordinates $(y_0, y_1)$, increase each index $i \in [y_0, y_1]$ by 1
+- When we run into a right boundary of some rectangle with $y$-coordinates $(y_0, y_1)$, decrease each index $i \in [y_0, y_1]$ by 1
+
+Then, for each $x$, we simply need to count the number of non-zero indices which corresponds to indices that are covered by at least one rectangle. How can we do this?
 
-describe
+Instead of counting the area covered by at least one rectangle, let's count the amount of space covered by no rectangles. We can subtract this amount from the total number of indices to get the value we want. 
 
-</IncompleteSection>
+We can use a Segment Tree that counts the number of occurences of the minimum value. Because the minimum value is at least zero (there can't be a negative number of rectangles at a position) the number of uncovered squares is equal to the number of squares with value 0.
 
 </Spoiler>
 
+### Implementation
+
+<LanguageSection>
+
+<CPPSection>
+
+```cpp
+#include <bits/stdc++.h>
+using namespace std;
+
+const int MAXX = 1e6;
+const int MAXN = 2 * MAXX + 1;
+ 
+struct Event {
+	int t, x, y0, y1;
+	// t = 1 for left bound, -1 for right bound
+	bool operator<(const Event &e) { return x < e.x; }
+};
+ 
+int N;
+vector<Event> E;
+	
+// segment tree
+pair<long long, long long> Tree[MAXN * 4];
+int Lazy[MAXN * 4];
```

Comment:
why in god's green earth are these variables capitalized

Reply:
because they're global

Reply:
what
what kind of convention do you follow

Reply:
does it really matter as long as what the variables do is clear

Reply:
honestly yes
if every single other solution (even benq's) variable names don't use this convention
you probably shouldn't yourself

Reply:
Benq's variable names should be used as a counter example for good naming conventions

Reply:
his macros are awful, yes
but at least his variable names are consistent
they use camelCase, or are just one letter
PascalCase is only for classes